docs/user/non-blocking-prefetch.md

```diff
@@ -0,0 +1,406 @@
+2. **Retry Logic**: Failed downloads are automatically retried with exponential backoff
+3. **Partial Download Cleanup**: Failed partial downloads are automatically cleaned up to prevent disk space exhaustion
+4. **Progress Reporting**: Regular status updates during long-running downloads
+5. **Concurrent Management**: Multiple downloads can be managed concurrently with proper resource coordination
+
+This documentation covers how to monitor and troubleshoot the prefetch system.
+
+## Understanding the Prefetch System
+
+The prefetch manager is automatically integrated into the Flight Control agent and operates transparently during device updates. When the agent receives a new device specification,
+it:
+
+1. **Identifies Required Images**: Determines which OS images and application artifacts need to be downloaded
+2. **Schedules Downloads**: Queues downloads in the background prefetch manager
+3. **Monitors Progress**: Tracks download progress and handles retries
+4. **Reports Status**: Provides status updates through device conditions
+5. **Manages Cleanup**: Automatically cleans up failed partial downloads
+
+### Prefetch Status Indicators
+
+You can monitor prefetch progress through device status conditions:
+
+```yaml
+status:
+  updated:
+    status: OutOfDate              # Update available but not yet ready
+  conditions:
+    - type: DeviceUpdating
+      status: "True"
+      reason: Preparing
+      message: "Downloading OS image in background"
+```
+
+## Download Management Features
+
+### Background Processing
+
+The prefetch manager operates in the background without blocking device operations:
+
+- **Non-blocking Downloads**: Images download while the device continues normal operations
+- **Queue Management**: Multiple downloads are queued and processed systematically
+- **Timeout Handling**: Configurable timeouts prevent hung downloads
+- **Error Handling**: Network errors and timeouts are treated as retryable conditions
+
+### Automatic Retry Logic
+
+When downloads fail, the system automatically retries with intelligent backoff:
+
+- **Exponential Backoff**: Retry delays increase progressively to avoid overwhelming networks
+- **Transient Error Recovery**: Network-related errors trigger automatic retries
+- **Persistent Error Reporting**: Non-retryable errors are reported immediately
+
+### Partial Download Cleanup
+
+To prevent disk space exhaustion from failed downloads:
+
+- **Automatic Cleanup**: Failed partial downloads are automatically removed
+- **Storage Protection**: Prevents accumulation of incomplete image layers
+- **Resource Conservation**: Frees up space for subsequent download attempts
+
+## Configuration and Timeout Settings
+
+### Pull Timeout Configuration
+
+The prefetch system respects the agent's configured pull timeout:
+
+```yaml
+# /etc/flightctl/config.yaml
+pull-timeout: 20m
```

Comment:
please reference pull-timeout as part of existing docs. we have been trying to keep all agent level config in this file.

https://github.com/flightctl/flightctl/blob/965d3832cd57c5ec0f1fff1a441c7afbf724ab12/docs/user/configuring-agent.md#agent-configyaml-configuration-file

Reply:
ok please reference `pull-timeout` as part of the existing docs with a link so users can easily find it?